### ISSUE TITLE
track history doesn't run?

### ISSUE THREAD

**Issue by solyaris** [piotrmurach/tty-prompt#50]
Hi Piotr!

thanks for your set of beautiful gems!

I'm trying to use your gem to a smart cli console (see: www.github.com/solyaris/rChatScript).
BTW, I'm using Ruby 2.4 on a Linux Ubuntu 14.04 server.

I have problems with track_history feature;

see this chunk of code:

```ruby
require 'tty-prompt'

DEFAULT_COLOR = :bright_blue
PROMPT_STRING =  '>> '
# ...

prompt = TTY::Prompt.new active_color: :green, help_color: :yellow #, track_history: true

puts "CTRL-C to stop"
loop do
  user_say = prompt.ask PROMPT_STRING

  bot_reply = client.volley(user_say, user: user).text

  puts pastel.decorate bot_reply, DEFAULT_COLOR
end
```

following what said here: https://github.com/piotrmurach/tty-prompt#36-track_history : 

>The prompts that accept line input such as multiline or ask provide history buffer that tracks all the lines entered during TTY::Prompt.new interactions. The history buffer provides previoius or next lines when user presses up/down arrows respectively. :

So, I would expect that using keyboard arrow keys, I could reuse previous line inserted with `prompt.ask`, instead, it seems to me that arrow keys are not "trapped":

```
>>  ciao
I don't know what to say.
>>  ^[[A^[[A^[[A^[[A^[[A
```

My aim is to use TTY:prompt having  features similar to those provided by http://bogojoker.com/readline/

Any idea?

Thanks a lot
giorgio

**Comment by piotrmurach** [comment]
Hi Giorgio,

Thanks for using the library! The `#ask` prompt editing capabilities are in the pipeline but not ready yet. Currently `#ask` works pretty much like `gets` call. I have only managed to roll out this feature to `#multiline` prompt in master. This hasn't been released yet and you can play with it using `examples/multiline.rb` file.

The `Reader#read_line` already provides the history navigation and text editing capabilities. However, I need to work more on it to account for line prompts etc... Also to bring many goodies over from Readline Editor it will require more work. I really want the `Reader` to be capable replacement for Readline and I intend to extract it as a standalone dependency. However, I cannot promise when this is going to be done.

I would encourage you to use `tty-prompt` with the view that the `#ask` will have history and editing features added in the near future. There is long outstanding issue about adding `#autocomplete` prompt which will be possible once I've done the editing bit.

**Comment by solyaris** [comment]
Thanks Piotr for  detailed answer!
So I wait a possible` Reader` replacement of `Readline` :-)
regards
giorgio

**Comment by piotrmurach** [comment]
Hey Giorgio, I've just released `v0.12.0` that replaces `ask` prompt with a new editing and history buffer features. You don't have to specify `:track_history` option if you want history buffer. Enjoy!

**Comment by solyaris** [comment]
Great Piotr!
sorry for delay (I'm so busy at work)
Asap I try new ask features and I'll for sure mention your beautiful work as dependency of a shell to interface ChatScript, I'll publish next weeks at github.com/solyaris/rChatScript

thanks again
respect
giorgio
